### PR TITLE
Updated to use latest Azure CLI

### PR DESCRIPTION
**Pull request by SharePointOscar**
Updated the contrib/azure-setup.sh file to use the latest Azure CLI version.  Fully tested.

Closes Docs: Azure Setup refers to outdated azure-cli 1.0 #5465

### CODE REVIEW

**Comment by mwhooker** [comment]
Thanks! Will let @boumenot weigh  in before merging

**Comment by SwampDragons** [comment]
ping @boumenot if you have no objections we'd like to merge this week for the 1.2.0 release

File: contrib/azure-setup.sh [hunk 1]
```diff
@@ -49,13 +49,13 @@ showhelp() {
...
-    azureversion=$(azure -v)
+    azureversion=$(az -v)
     if [ $? -eq 0 ]; then
         found=$((found + 1))
         echo "Found azure-cli version: $azureversion"
     else
-        echo "azure-cli is missing. Please install azure-cli from"
-        echo "https://azure.microsoft.com/en-us/documentation/articles/xplat-cli-install/"
+        echo "azure cli is missing. Please install azure cli from"
+        echo "https://az.microsoft.com/en-us/documentation/articles/xplat-cli-install/"
```
Comment (boumenot) [inline contrib/azure-setup.sh]: This isn't the XPlat CLI, this it the Azure CLI 2.0.

File: contrib/azure-setup.sh [hunk 2]
```diff
@@ -49,13 +49,13 @@ showhelp() {
...
 
-    azureversion=$(azure -v)
+    azureversion=$(az -v)
     if [ $? -eq 0 ]; then
         found=$((found + 1))
         echo "Found azure-cli version: $azureversion"
     else
-        echo "azure-cli is missing. Please install azure-cli from"
-        echo "https://azure.microsoft.com/en-us/documentation/articles/xplat-cli-install/"
+        echo "azure cli is missing. Please install azure cli from"
```
Comment (boumenot) [inline contrib/azure-setup.sh]: I would point people in the Azure portal, and ask them to use Cloud Shell.

File: contrib/azure-setup.sh [hunk 3]
```diff
@@ -158,7 +169,7 @@ createApplication() {
 
 createServicePrincipal() {
     echo "==> Creating service principal"
-    # Azure CLI 0.10.2 introduced a breaking change, where appId must be supplied with the -a switch
+    # az CLI 0.10.2 introduced a breaking change, where appId must be supplied with the -a switch
```
Comment (boumenot) [inline contrib/azure-setup.sh]: Comment does not match the code (XPlat CLI -> Azure CLI 2.0).

File: contrib/azure-setup.sh (hunk 2 above)
Comment (SharePointOscar) [reply contrib/azure-setup.sh]: Sounds good, added verbiage and link for user to learn about the Cloud Shell

File: contrib/azure-setup.sh (hunk 1 above)
Comment (SharePointOscar) [reply contrib/azure-setup.sh]: My mistake, I changed the link to the appropriate one.

File: contrib/azure-setup.sh (hunk 3 above)
Comment (SharePointOscar) [reply contrib/azure-setup.sh]: Changed to appropriate CLI naming.

File: contrib/azure-setup.sh [hunk 4]
```diff
@@ -169,7 +170,7 @@ createApplication() {
 
 createServicePrincipal() {
     echo "==> Creating service principal"
-    # az CLI 0.10.2 introduced a breaking change, where appId must be supplied with the -a switch
+    # Azure CLI 0.10.2 introduced a breaking change, where appId must be supplied with the -a switch
```
Comment (boumenot) [inline contrib/azure-setup.sh]: I would kill this whole line because you're using CLI 2.0 instead of XPlatCLI.

Comment (SharePointOscar) [reply contrib/azure-setup.sh]: Thank you @boumenot I will clean that up.  In addition, I thought of enhancing the script further to provide an option to remove all of the resources the script creates when running with the 'setup' parameter.  So we could have a 

```bash
./azure-setup.sh destroy
```

**Pull request merged**